apis/v1beta1/shared_types.go

```diff
@@ -217,6 +217,10 @@ const (
 	// compatible Listeners whose Hostname matches the route
 	RouteReasonNoMatchingListenerHostname RouteConditionReason = "NoMatchingListenerHostname"
 
+	// This reason is used with the "Accepted" condition when a value for an Enum
+	// is not recognized.
+	RouteReasonInvalidEnumValue RouteConditionReason = "InvalidEnumValue"
```

Comment:
Should we make this more generic? Maybe just "UnknownValue" or "UnsupportedValue"?

Reply:
Yeah, that makes sense, I'll update.